awx/main/dispatch/publish.py

```diff
@@ -83,6 +84,11 @@ def apply_async(cls, args=None, kwargs=None, queue=None, uuid=None, **kw):
                     'kwargs': kwargs,
                     'task': cls.name
                 }
+                correlation_id = GuidMiddleware.get_guid()
```

Comment:
When `.apply_async()` is called, get the latest GUID from the current thread, *or* generate a new one on the fly (in case it's called from some other non-HTTP request context, like the task manager).